contracts/Items/DEFYForge.sol

```diff
@@ -0,0 +1,241 @@
+        external
+        view
+        returns (bytes32)
+    {
+        require(index < getPrintJobsCount(), "DEFYForge: index out of bounds");
+        return printJobIds[index];
+    }
+
+    /**
+     * @dev Returns the number of print jobs created by an operative.
+     * @return the number of print jobs
+     */
+    function getPrintJobsCount() public view returns (uint256) {
+        return printJobIds.length;
+    }
+
+    /**
+     * @notice Returns the print job information for a given identifier.
+     * @return the print job structure information
+     */
+    function getPrintJob(bytes32 printJobId)
+        public
+        view
+        returns (PrintJob memory)
+    {
+        return printJobs[printJobId];
+    }
+
+    /**
+     * @dev Returns the number of print jobs associated to a operative.
+     * @return the number of print jobs
+     */
+    function getPrintJobsCountByOperative(address _operative)
+        external
+        view
+        returns (uint256)
+    {
+        return operativesPrintJobCount[_operative];
+    }
+
+    /**
+     * @dev Returns the print job id at the given index.
+     * @return the print job id
+     */
+    function getVestingIdAtIndex(uint256 index)
+        external
+        view
+        returns (bytes32)
+    {
+        require(index < getPrintJobsCount(), "DEFYForge: index out of bounds");
+        return printJobIds[index];
+    }
+
+    /**
+     * @notice Returns the print job information for a given operative and index.
+     * @return the print job structure information
+     */
+    function getPrintJobByAddressAndIndex(address operative, uint256 index)
+        external
+        view
+        returns (PrintJob memory)
+    {
+        return
+            getPrintJob(computePrintJobIdForAddressAndIndex(operative, index));
+    }
+
+    /**
+     * @dev Computes the next print job identifier for a given operative address.
+     */
+    function computeNextPrintJobIdForOperative(address operative)
+        public
+        view
+        returns (bytes32)
+    {
+        return
+            computePrintJobIdForAddressAndIndex(
+                operative,
+                operativesPrintJobCount[operative]
+            );
+    }
+
+    /**
+     * @dev Returns the last print job for a given operative address.
+     */
+    function getLastPrintJobForOperative(address operative)
+        external
+        view
+        returns (PrintJob memory)
+    {
+        return
+            printJobs[
+                computePrintJobIdForAddressAndIndex(
+                    operative,
+                    operativesPrintJobCount[operative] - 1
+                )
+            ];
+    }
+
+    /**
+     * @dev Computes the print job identifier for an address and an index.
+     */
+    function computePrintJobIdForAddressAndIndex(
+        address operative,
+        uint256 index
+    ) public pure returns (bytes32) {
+        return keccak256(abi.encodePacked(operative, index));
+    }
```

Comment:
We can just use uint256s for print job ids and increment the counter as jobs are created